extensions/meme_rater.py

```diff
@@ -0,0 +1,50 @@
+import os
+import hikari, lightbulb
+from PIL import Image
+import requests
+from snark import model
+from google.generativeai.generative_models import GenerativeModel
+
+plugin = lightbulb.Plugin("MemeRater")
+
+MEMES_CHANNEL_ID = os.environ["MEME_CHANNEL_ID"]
+
+#Make this envs if you want
+MEME_RATE_PROMPT = "Rate this meme out of 10, with 10 being the funniest. Rate is solely on how funny a bunch of computer science nerds would find it. ONLY Return an integer."
+MINIMUM_MEME_RATING_TO_NOT_DELETE = 6
+
+async def get_meme_rating(image_url:str, model:GenerativeModel = model):
+    response = model.generate_content(contents=[MEME_RATE_PROMPT, Image.open(requests.get(image_url).raw)])
+    return response.text
+
+
+@plugin.listener(hikari.GuildMessageCreateEvent)
+async def main(event: hikari.GuildMessageCreateEvent) -> None:
+    if event.channel_id == MEMES_CHANNEL_ID:
+        # Don't handle messages from bots or without content
+        if event.is_bot:
+            return
+        for attachment in event.message.attachments:
```

Comment:
Does this include images sent by url instead of direct file upload?

Reply:
Direct file upload only (for now)